CBPs/cbp-0002.md

```diff
@@ -0,0 +1,632 @@
+
+  However, in the Dex and lending scenarios of Ethereum, Approve is often accompanied by the possibility of simultaneous operation of two tokens. Approve can avoid the repeated payment problem which transaction brought about, has a good supplementary use scenario for transfer.
+
+- Solution: Approve/transferFrom should be retained
+
+8. TransferAndCall vs Receiver Notify
+- Question: which option is more suitable
+- Consideration:
+  - Notify can meet the basic notification needs. Although it cannot support better flexibility, it is sufficient to meet the transfer scenario
+  - TransferAndCall provides better flexibility, but it depends on the transfer caller to fully understand the method and parameters corresponding to the call, which is not needed for most transfer scenarios
+  - For security reasons, do not call the canister of the location inside the canister,[why?Inter-canister calls](https://www.joachim-breitner.de/blog/788-How_to_audit_an_Internet_Computer_canister)
+  
+- Solution: 
+  - Neither is supported
+
+9. ApproveAndCall VS TransferAndCall
+
+- Question: We compare ApproveAndCall and TransferAndCall. ApproveAndCall and TransferAndCall are two sets of non-atomic operations, there is no difference essentially. Which one should be retained?
+
+- Consideration: In some scenarios, when multiple Tokens need to be transferred at the same time, TransferAndCall can not meet such needs. After approval, execute transferFrom in the final call to pay multiple tokens at once
+- For security reasons like Q8
+
+- Solution: call is not supported
+
+10. Nonce
+
+- Question: What Is Nonce In Ethereum? every transaction has a nonce. The nonce is the number of transactions sent from a given address. Each time you send a transaction, the nonce increases by 1. There are rules about what transactions are valid, and the nonce is used to enforce some of these rules. Does Dfinity's token standard need to incorporate a nonce mechanism?
+
+- Consideration:If the nonce mechanism is supported, it can prevent users from repeatedly submitting transactions
+
+- Solution: Support nonce mechanism
+
+#### What other functions does the Dfinity Fungible Token standard need to implement?
+ 
+**Information self-describing**
+
+Etherscan, MyEthereumWallet, Imtoken, TokenPocket, Dapp all have more information requirements for ERC20, such as Logo, introduction, white paper, social media, official website, contact information, etc. Each of them that needs this information needs to be maintained independently, so information appears Inconsistent. It is necessary to solve this problem through the design of <strong>[Dfinity Fungible Token Standard]</strong>
```

Comment:
So if you match the Rosetta block fetching interface, you should get the ic.rocks token explorer working for you for free.

Reply:
This is something I am not in a hurry to implement. Because I think the token standard is more important community consensus, when the community consensus is reached, support resetta is easy

Reply:
So I'm not suggesting supporting Rosetta in the standard. But having a block fetching interface like ICP allows CEXes to use the token and more importantly allows you to have backups of the transaction logs.

Reply:
Sounds good, and I think this issue can be put to the interface consensus stage to discuss and make the decision, which does make sense.